### ISSUE TITLE
gather futures from multiple executions

### ISSUE THREAD

**Issue by omerb01** [pywren/pywren-ibm-cloud#105]
an issue that I noticed from the last commits,
PyWren can't run anymore multiple executions like in this example:
```python
def simple_map_function(x, y):
    return x + y

pw = pywren.ibm_cf_executor()
iterdata = [[1, 1], [2, 2]]
pw.map(simple_map_function, iterdata)
iterdata = [[3, 3], [4, 4]]
pw.map(simple_map_function, iterdata)
result = pw.get_result()
```
cause the result is the result of the last execution instead of all of the above.

**Comment by gilv** [comment]
@JosepSampe any input? thanks

**Comment by gilv** [comment]
@omerb01 is this issue somehow related this fix? https://github.com/pywren/pywren-ibm-cloud/pull/104

**Comment by omerb01** [comment]
@gilv no, this is only a simple fix for `map()` call inside `map_reduce()` method

**Comment by JosepSampe** [comment]
@omerb01 I was doing some tests with this because the approach you suggested may produce inconsistent results when PyWren is used in notebooks. So finally I thought that the best approach is to keep internally only the futures of the last invocation.

For example, in the metabolomics notebook, you create the PyWren executor in one block:
`pw = pywren.ibm_cf_executor()`
and then use the same executor for different invocations in different blocks:
```
iterdata = [[f'{bucket_name}/{input_data["ds"]}', parse_spectrum_line]]
pw.map_reduce(parse_txt, iterdata, reduce_chunks, chunk_size=64 * 1024 ** 2)
spectra = pw.get_result()
```
If you run multiple times this second block without storing the futures in a variable, you will get inconsistent results. So, I think if someone wants to use same executor for multiple invocations, the best approach is to use futures (as it is now used in the notebook due this issue).
```
pw = pywren.ibm_cf_executor()
iterdata = [[1, 1], [2, 2]]
ftrs1 = pw.map(simple_map_function, iterdata)
iterdata = [[3, 3], [4, 4]]
ftrs2 = pw.map(simple_map_function, iterdata)
result = pw.get_result(ftrs1.extend(ftrs2))
```

**Comment by omerb01** [comment]
@JosepSampe thanks for the explanation.
In my opinion, this feature is useful cause it makes the code more clean and intuitive as well.
I suggest that `self.futures` of `ibm_cf_executor` struct would be appended by every PyWren execution and when you will operate `get_result()` method, `self.futures` will be cleared.
what do you think?

**Comment by JosepSampe** [comment]
@omerb01 sounds good

**Comment by JosepSampe** [comment]
Already fixed in #106
